pkgs/development/tools/java/jprofiler/default.nix

```diff
@@ -0,0 +1,100 @@
+{ stdenv
+, lib
+, fetchurl
+, makeWrapper
+, makeDesktopItem
+, undmg
+, jdk11 }:
+
+let
+  inherit (stdenv.hostPlatform) system;
+  pname = "jprofiler";
+
+  # 11.1.4 is the last version which can be unpacked by undmg
+  # See: https://github.com/matthewbauer/undmg/issues/9
+  version = if stdenv.isLinux then "13.0.2" else "11.1.4";
+  nameApp = "JProfiler";
+
+  meta = with lib; {
+    description = "JProfiler's intuitive UI helps you resolve performance bottlenecks";
+    longDescription = ''
+      JProfiler's intuitive UI helps you resolve performance bottlenecks,
+      pin down memory leaks and understand threading issues.
+    '';
+    homepage = "https://www.ej-technologies.com/products/jprofiler/overview.html";
+    license = licenses.unfree;
+    maintainers = with maintainers; [ catap ];
+  };
+
+  src = fetchurl (if stdenv.isLinux then {
+    url = "https://download-gcdn.ej-technologies.com/jprofiler/jprofiler_linux_${builtins.replaceStrings ["."] ["_"]  version}.tar.gz";
+    sha256 = "sha256-x9I7l2ctquCqUymtlQpFXE6+u0Yg773qE6MvAxvCaEE=";
+  } else {
+    url = "https://download-gcdn.ej-technologies.com/jprofiler/jprofiler_macos_${builtins.replaceStrings ["."] ["_"]  version}.dmg";
+    sha256 = "sha256-WDMGrDsMdY1//WMHgr+/YKSxHWt6A1dD1Pd/MuDOaz8=";
+  });
+
+
+  srcIcon = fetchurl {
+    url = "https://www.ej-technologies.com/assets/content/header-product-jprofiler@2x-24bc4d84bd2a4eb641a5c8531758ff7c.png";
+    sha256 = "sha256-XUmuqhnNv7mZ3Gb4A0HLSlfiJd5xbCExVsw3hmXHeVE=";
+  };
+
+  desktopItem = makeDesktopItem {
+    name = pname;
+    exec = pname;
+    icon = pname;
+    comment = meta.description;
+    desktopName = nameApp;
+    genericName = "Java Profiler Tool";
+    categories = [ "Development" ];
+  };
+
+  linux = stdenv.mkDerivation {
+    inherit pname version src;
+
+    nativeBuildInputs = [ makeWrapper ];
+
+    installPhase = ''
+      cp -r . $out
```

Comment:
You shouldn't do that. As I've mentioned earlier, this install _**a lot**_ of unnecessary files. If you think that we should keep docs or examples, do this in a separate derivation [output](https://nixos.org/manual/nixpkgs/stable/#sec-multiple-outputs-). Also make sure that there are no redundant libraries for other architectures/platforms.

Reply:
The only things that I can remove is a few libraries for unsupported by nixos architectures. It saves something like 10%-15% of diskspace — nothing from my point of view.

"samples" are available via UI
<img width="1168" alt="Screenshot 2022-07-01 at 01 24 34" src="https://user-images.githubusercontent.com/37775/176794742-e2bc85c7-d489-43f7-ad7e-b58fbc1c2250.png">
 of JProfiler => I really don't think that it is a good idea to remove them:
<img width="1168" alt="Screenshot 2022-07-01 at 01 24 39" src="https://user-images.githubusercontent.com/37775/176794761-298babdd-0dd7-4d2f-bf81-e20f157a3241.png">

Reply:
It's not the issue of occupied space (although, this could be annoying given that Hydra could rebuild this package frequently and waste a lot of unneeded space across multiple derivations). The issue here is the files that are not supposed to be there because they are functionally meaningless, as the case with libraries and executable which will never run. 

As for examples and documentations, I still think that a separate output would be preferable. Your justification to use examples from the GUI makes sense too. I'll leave it up to you.

Reply:
@azahi see example like intellij-idea which. It is quite standalone tool.